#################################################################################
#   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.          #
#                                                                               #
#   Licensed under the Apache License, Version 2.0 (the "License").             #
#   You may not use this file except in compliance with the License.            #
#   You may obtain a copy of the License at                                     #
#                                                                               #
#       http://www.apache.org/licenses/LICENSE-2.0                              #
#                                                                               #
#   Unless required by applicable law or agreed to in writing, software         #
#   distributed under the License is distributed on an "AS IS" BASIS,           #
#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.    #
#   See the License for the specific language governing permissions and         #
#   limitations under the License.                                              #
#################################################################################

controller_manager:
  ros__parameters:
    update_rate: 10  # Hz

    joint_state_controller:
      type: joint_state_controller/JointStateController

    left_rear_wheel_velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    right_rear_wheel_velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    left_front_wheel_velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    right_front_wheel_velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    left_steering_hinge_position_controller:
      type: position_controllers/JointGroupPositionController

    right_steering_hinge_position_controller:
      type: position_controllers/JointGroupPositionController

left_rear_wheel_velocity_controller:
  ros__parameters:
    joints:
      - left_rear_wheel_joint
    interface_name: velocity
    pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}

right_rear_wheel_velocity_controller:
  ros__parameters:
    joints:
      - right_rear_wheel_joint
    interface_name: velocity
    pid: {p: 1.0, i: 0.0, d: 0.0, i_clamp: 0.0}


left_front_wheel_velocity_controller:
  ros__parameters:
    joints:
      - left_front_wheel_joint
    interface_name: velocity
    pid: {p: 0.5, i: 0.0, d: 0.0, i_clamp: 0.0}


right_front_wheel_velocity_controller:
  ros__parameters:
    joints:
      - right_front_wheel_joint
    interface_name: velocity
    pid: {p: 0.5, i: 0.0, d: 0.0, i_clamp: 0.0}


left_steering_hinge_position_controller:
  ros__parameters:
    joints:
      - left_steering_hinge_joint
    interface_name: position
    pid: {p: 1.0, i: 0.0, d: 0.5}


right_steering_hinge_position_controller:
  ros__parameters:
    joints:
      - right_steering_hinge_joint
    interface_name: position
    pid: {p: 1.0, i: 0.0, d: 0.5}
